e2e_playwright/st_pills.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import streamlit as st
+
+st.header("Pills - standard")
+
+if st.checkbox("Set default values", value=False):
+    st.session_state.default_pills = ["🧰 General widgets", "📊 Charts", "🧊 3D"]
+else:
+    st.session_state.default_pills = []
+
+default = st.session_state.default_pills
+
+pills_options = [
+    "🧰 General widgets",
+    "📊 Charts",
+    "🌇 Images",
+    "🎥 Video",
+    "📝 Text",
+    "🗺️ Maps & geospatial",
+    "🧮 Dataframes & tables",
+    "🧬 Molecules & genes",
+    "🪢 Graphs",
+    "🧊 3D",
+    "✏️ Code & editors",
+    "📃 Page navigation",
+    "🔐 Authentication",
+    "🎨 Style & layout",
+    "🛠️ Developer tools",
+    "🏗️ App builders",
+    "🔌 Integrations with other tools",
+    "📦 Collections of components",
+]
+selection = st.pills(
+    "Select some options",
+    pills_options,
+    key="pills",
+    selection_mode="multi",
+    default=default,
+    help="This is for choosing options",
+)
+st.write(f"Multi selection: {selection}")
+
+
+st.header("Pills - with icons")
+selection = st.pills(
+    "Select a single option",
+    options=[0, 1, 2, 3],
+    icons=[
+        ":material/add:",
+        ":material/zoom_in:",
+        ":material/zoom_out:",
+        ":material/zoom_out_map:",
+    ],
```

Comment:
yeah you can but then the styling is a little bit different though (first example corresponds to the code above where the icons are passed for the icons argument; the second one is what you are referring to):

<img width="802" alt="Screenshot 2024-09-13 at 16 42 14" src="https://github.com/user-attachments/assets/5e17bafb-2dcf-4f3d-8335-500421ec531f">

From product-side we did not want to "autodetect" whether the option is just an icon; so the user has to make the active choice.

Reply:
And what would be the recommended usage when combined with text?

```python
selection = st.pills(
    "Select some options",
    options=[
         "General widgets",
          "Charts",
          "Images",
    ],
    icons=[
        "🧰",
        "📊",
        "🌇",
     ]
)
```

Instead of

```python
selection = st.pills(
    "Select some options",
    options=[
    "🧰 General widgets",
    "📊 Charts",
    "🌇 Images"
    ]
)
```

Reply:
> From product-side we did not want to "autodetect" whether the option is just an icon;

But why would you have to autodetect it? Because you apply different styling? The two examples in the screenshot look the same to me

Reply:
they are almost identical but the one uses DynamicIcon and the other StreamlitMarkdown. If you look at the properties, they are slightly different. Recommendation is to use the icons argument for icons similar to how it is for `st.button` etc